components/camel-zeebe/src/main/java/org/apache/camel/component/zeebe/ZeebeConsumer.java

```diff
@@ -0,0 +1,122 @@
+import org.apache.camel.Processor;
+import org.apache.camel.component.zeebe.internal.OperationName;
+import org.apache.camel.component.zeebe.model.JobWorkerMessage;
+import org.apache.camel.support.DefaultConsumer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ZeebeConsumer extends DefaultConsumer {
+    private static final Logger LOG = LoggerFactory.getLogger(ZeebeConsumer.class);
+
+    private final ZeebeEndpoint endpoint;
+
+    private JobWorker jobWorker;
+
+    private ObjectMapper objectMapper = new ObjectMapper();
+
+    public ZeebeConsumer(ZeebeEndpoint endpoint, Processor processor) throws CamelException {
+        super(endpoint, processor);
+        this.endpoint = endpoint;
+    }
+
+    @Override
+    protected void doStart() throws Exception {
+        super.doStart();
+
+        final OperationName operationName = getEndpoint().getOperationName();
+        switch (operationName) {
+            case REGISTER_JOB_WORKER:
+                if (getEndpoint().getJobKey() == null) {
+                    LOG.error("Missing JobKey");
+                    throw new CamelException("Missing JobKey");
+                }
+                jobWorker = getEndpoint().getZeebeService().registerJobHandler(new ConsumerJobHandler(),
+                        getEndpoint().getJobKey(), getEndpoint().getTimeout());
+                break;
+            default:
+                LOG.error("Invalid Operation %s", operationName.value());
+                throw new CamelException(String.format("Invalid Operation for Consumer %s", operationName.value()));
+        }
+    }
+
+    @Override
+    protected void doStop() throws Exception {
+        super.doStop();
+
+        if (jobWorker != null && jobWorker.isOpen()) {
+            jobWorker.close();
+        }
+    }
+
+    private class ConsumerJobHandler implements JobHandler {
+        @Override
+        public void handle(JobClient client, ActivatedJob job) throws Exception {
+            final Exchange exchange = createExchange(true);
+            final Message in = exchange.getIn();
+
+            JobWorkerMessage message = new JobWorkerMessage();
+            message.setKey(job.getKey());
+            message.setType(job.getType());
+            message.setCustomHeaders(job.getCustomHeaders());
+            message.setProcessInstanceKey(job.getProcessInstanceKey());
+            message.setBpmnProcessId(job.getBpmnProcessId());
+            message.setProcessDefinitionVersion(job.getProcessDefinitionVersion());
+            message.setProcessDefinitionKey(job.getProcessDefinitionKey());
+            message.setElementId(job.getElementId());
+            message.setElementInstanceKey(job.getElementInstanceKey());
+            message.setWorker(job.getWorker());
+            message.setRetries(job.getRetries());
+            message.setDeadline(job.getDeadline());
+            message.setVariables(job.getVariablesAsMap());
+
+            LOG.info(job.toJson());
```

Comment:
A few things:

1. Maybe add a message explaining what is being logged.
2. Also, seems like it would be better to use a lower debug level (`debug`, maybe?).